### DISCUSSION TITLE
"继承引擎类功能"是否是不建议使用的？

### DISCUSSION THREAD

**Discussion by watsonsong** [Tencent/puerts#1346] [Q&A]
https://github.com/Tencent/puerts/commit/23b27e8e35b8b63cc32c7e65800efff9bd718695
看到之前文档的一个更新，想了解一下是否对于新项目就不建议基于“继承引擎类”功能来设计开发了？

**Comment by chexiongsheng** [comment]
这功能使用没问题，但不要滥用，比如一段逻辑明明调用方只有ts，非得继承一个UE类来写。
正确的做法是尽可能的让引擎和脚本各访问各的，尽在边界部分使用该功能。
滥用导致过多的跨语言访问影响到性能。
